lib/templates/middleware/context.js

```diff
@@ -6,7 +6,15 @@ import { getBackendLayout, getFrontendLayout } from '~typo3/lib/layouts'
 const setContext = (context, response) => {
   if (response && response.data) {
     if (response.data.redirectUrl) {
-      return context.redirect(response.data.redirectUrl)
+      if (process.server) {
+        return context.redirect(
+          response.data.statusCode || 200,
+          response.data.redirectUrl
+        )
+      } else if (process.client) {
+        window.location.href = response.data.redirectUrl
+        return
```

Comment:
here we can add `return context.next(false)` to avoid additional route change. Please check also it window is not undefined